bcda/auth/models.go

```diff
@@ -63,3 +68,55 @@ func GetACOByClientID(clientID string) (models.ACO, error) {
 	}
 	return aco, err
 }
+
+/*
+ GenerateSystemKeyPair creates a keypair for a system. The public key is saved to the database and the private key is returned.
+*/
+func GenerateSystemKeyPair(systemID uint) (string, error) {
+	db := database.GetGORMDbConnection()
+	defer database.Close(db)
+
+	var system models.System
+	err := db.Preload("EncryptionKeys").First(&system, systemID).Error
```

Comment:
i think `Find()` makes more sense semantically here but i know we've talked this to death already